src/command_modules/azure-cli-container/azure/cli/command_modules/container/custom.py

```diff
@@ -40,29 +41,78 @@ def create_container(client,
-    """"Create a container group. """
+                     registry_password=None,
+                     azure_file_volume_share_name=None,
+                     azure_file_volume_account_name=None,
+                     azure_file_volume_account_key=None,
+                     azure_file_volume_mount_path=None):
+    """Create a container group. """
 
-    container_resource_requirements = None
-    if cpu is not None or memory is not None:
+    ports = ports or []
+
+    container_resource_requirements = create_resource_requirements(cpu=cpu, memory=memory)
+
+    image_registry_credentials = create_image_registry_credentials(registry_login_server=registry_login_server,
+                                                                   registry_username=registry_username,
+                                                                   registry_password=registry_password,
+                                                                   image=image)
+
+    command = shlex.split(command_line) if command_line else None
+
+    azure_file_volume = create_azure_file_volume(azure_file_volume_share_name=azure_file_volume_share_name,
+                                                 azure_file_volume_account_name=azure_file_volume_account_name,
+                                                 azure_file_volume_account_key=azure_file_volume_account_key)
+
+    azure_file_volume_mount = create_azure_file_volume_mount(azure_file_volume=azure_file_volume,
+                                                             azure_file_volume_mount_path=azure_file_volume_mount_path)
+
+    container = Container(name=name,
+                          image=image,
+                          resources=container_resource_requirements,
+                          command=command,
+                          ports=[ContainerPort(port=p) for p in ports],
+                          environment_variables=environment_variables,
+                          volume_mounts=azure_file_volume_mount)
+
+    cgroup_ip_address = create_ip_address(ip_address, ports)
+
+    cgroup = ContainerGroup(location=location,
+                            containers=[container],
+                            os_type=os_type,
+                            restart_policy=restart_policy,
+                            ip_address=cgroup_ip_address,
+                            image_registry_credentials=image_registry_credentials,
+                            volumes=azure_file_volume)
+
+    return client.container_groups.create_or_update(resource_group_name, name, cgroup)
+
+
+def create_resource_requirements(cpu, memory):
+    """Create resource requirements. """
+    if cpu or memory:
         container_resource_requests = ResourceRequests(memory_in_gb=memory, cpu=cpu)
-        container_resource_requirements = ResourceRequirements(requests=container_resource_requests)
+        return ResourceRequirements(requests=container_resource_requests)
+    return None
```

Comment:
Remove this. See reason in later part of the review.

Reply:
Good to know